Clean up build process for tests.

kevingranade opened the pull request:
Just a little adjustment to the build logic.  Makes an object archive as a prerequisite for recursing into the tests directory instead of grabbing the object files directly.
This in principle should be slightly faster, but it's still quite slow.
Warning, on my system my obj dir is almost 1GB, and the archive this produces is over 500MB, and the individual test executables are.... oddly either 135MB or 5MB, weird.

BevapDin commented on tests/Makefile:
```diff
@@ -1,30 +1,29 @@
+# We're using the fairly typical convention that any file ending in _test.cpp
+# is a test executable.
+SOURCES = $(wildcard *_test.cpp)
+OBJS = $(TEST_SOURCES:%.cpp=$(ODIR)/%.o)
```
It appears `TEST_SOURCES` here and in the line below must be renamed as well. Otherwise, `OBJS` and `TESTS` end up empty.

BevapDin commented:
Running the test cases via make doesn't really work, it tries to start a single program like this:
```
LD_LIBRARY_PATH=/usr/local/lib ./line_test shadowcasting_test map_iterator_test player_test
```
The test programs need to be run separately.

player_test needs to be run from the main folder as it requires at least the Lua startup files (classdefintions and autoexec) and it also complains about missing color definitions (main.cpp calls `init_colors`, that may be needed in the test file as well).

The other tests work fine.

> and the individual test executables are.... oddly either 135MB or 5MB, weird.

I guess map_iterator_test is only 5MB because it does not include any of the main game data (no game class, no map, no player etc.)

kevingranade commented:
I had actually forgotten the check target was in there.
I'm fixing those two issues, but please don't avoid merging because of the player test, both the test and the code in player it's testing need a complete overhaul to avoid needing to generate the whole game world to test one thing.  That should help with the wacky file sizes too.

kevingranade commented:
Fixed the file enumeration bug, fixed the check target, some more general cleanup, disabled the player test since it's broken, and dropped the number of iterations for the perf tests so they'll run fast.
